exporter/datadogexporter/internal/translator/ttlcache_test.go

```diff
@@ -24,21 +24,91 @@ func newTestCache() *ttlCache {
 	cache := newTTLCache(1800, 3600)
 	return cache
 }
-func TestPutAndGetDiff(t *testing.T) {
+
+func TestMonotonicDiffUnknownStart(t *testing.T) {
+	startTs := uint64(0) // equivalent to start being unset
+	prevPts := newTestCache()
+	_, ok := prevPts.MonotonicDiff("test", []string{}, startTs, 1, 5)
+	assert.False(t, ok, "expected no diff: first point")
+	_, ok = prevPts.MonotonicDiff("test", []string{}, startTs, 0, 0)
+	assert.False(t, ok, "expected no diff: old point")
+	_, ok = prevPts.MonotonicDiff("test", []string{}, startTs, 2, 2)
+	assert.False(t, ok, "expected no diff: new < old")
+	dx, ok := prevPts.MonotonicDiff("test", []string{}, startTs, 3, 4)
+	assert.True(t, ok, "expected diff: no startTs, old >= new")
+	assert.Equal(t, 2.0, dx, "expected diff 2.0 with (0,2,2) value")
+}
+
+func TestDiffUnknownStart(t *testing.T) {
+	startTs := uint64(0) // equivalent to start being unset
+	prevPts := newTestCache()
+	_, ok := prevPts.Diff("test", []string{}, startTs, 1, 5)
+	assert.False(t, ok, "expected no diff: first point")
+	_, ok = prevPts.Diff("test", []string{}, startTs, 0, 0)
+	assert.False(t, ok, "expected no diff: old point")
+	dx, ok := prevPts.Diff("test", []string{}, startTs, 2, 2)
+	assert.True(t, ok, "expected diff: no startTs, not monotonic")
+	assert.Equal(t, -3.0, dx, "expected diff -3.0 with (0,1,5) value")
+	dx, ok = prevPts.Diff("test", []string{}, startTs, 3, 4)
+	assert.True(t, ok, "expected diff: no startTs, old >= new")
+	assert.Equal(t, 2.0, dx, "expected diff 2.0 with (0,2,2) value")
+}
+
+func TestMonotonicDiffKnownStart(t *testing.T) {
+	startTs := uint64(1)
 	prevPts := newTestCache()
-	_, ok := prevPts.putAndGetDiff("test", []string{}, 1, 5)
-	// no diff since it is the first point
-	assert.False(t, ok)
-	_, ok = prevPts.putAndGetDiff("test", []string{}, 0, 0)
-	// no diff since ts is lower than the stored point
-	assert.False(t, ok)
-	_, ok = prevPts.putAndGetDiff("test", []string{}, 2, 2)
-	// no diff since the value is lower than the stored value
-	assert.False(t, ok)
-	dx, ok := prevPts.putAndGetDiff("test", []string{}, 3, 4)
-	// diff with the most recent point (2,2)
-	assert.True(t, ok)
-	assert.Equal(t, 2.0, dx)
+	_, ok := prevPts.MonotonicDiff("test", []string{}, startTs, 1, 5)
+	assert.False(t, ok, "expected no diff: first point")
+	_, ok = prevPts.MonotonicDiff("test", []string{}, startTs, 0, 0)
+	assert.False(t, ok, "expected no diff: old point")
+	_, ok = prevPts.MonotonicDiff("test", []string{}, startTs, 2, 2)
+	assert.False(t, ok, "expected no diff: new < old")
+	dx, ok := prevPts.MonotonicDiff("test", []string{}, startTs, 3, 4)
+	assert.True(t, ok, "expected diff: same startTs, old >= new")
+	assert.Equal(t, 2.0, dx, "expected diff 2.0 with (0,2,2) value")
+
+	startTs = uint64(4) // simulate reset
+	_, ok = prevPts.MonotonicDiff("test", []string{}, startTs, 4, 8)
+	assert.False(t, ok, "expected no diff: reset with unknown start")
```

Comment:
Since the timestamp isn't 0, the start is known here, isn't it?

Reply:
Spec says

> When StartTimeUnixNano equals TimeUnixNano, a new unbroken sequence of observations begins with a reset at an unknown start time.

that's what I meant with 'unknown start' here. Do you think I could rephrase it in some way to make it clearer?

Reply:
I see now, maybe add to the comment on the line above: `startTs = uint64(4) // simulate reset with startTs = ts`, and/or `_, ok = prevPts.MonotonicDiff("test", []string{}, startTs, startTs, 8)` to be explicit?

Reply:
done!